### PR TITLE
Refactor configgrpc for compression methods

### PR DESCRIPTION
**Pull request by hyunuk**
**Description:**
- use `middleware.CompressionType` instead of grpc's own constants.
- Not to use a map to validate compression keys.
- added tests for each compression methods.
- throw an error for zlib and deflate, which only be supported in confighttp

**Link to tracking Issue:**
#4623

### CODE REVIEW

File: config/configgrpc/configgrpc.go [hunk 1]
```diff
@@ -373,14 +356,18 @@ func (gss *GRPCServerSettings) ToServerOption(host component.Host, settings comp
...
 }
 
-// GetGRPCCompressionKey returns the grpc registered compression key if the
-// passed in compression key is supported, and CompressionUnsupported otherwise.
-func GetGRPCCompressionKey(compressionType string) string {
-	compressionKey := strings.ToLower(compressionType)
-	if encodingKey, ok := gRPCCompressionKeyMap[compressionKey]; ok {
-		return encodingKey
+// GetGRPCCompressionName returns compression name registered in grpc.
+func GetGRPCCompressionName(compressionType middleware.CompressionType) (string, error) {
```
Comment (bogdandrutu) [inline config/configgrpc/configgrpc.go:360]: Why is this public? Can we make this private?

**Review by Aneurysm9 — APPROVED** [review]

File: config/configgrpc/configgrpc.go [hunk 2]
```diff
@@ -83,7 +66,7 @@ type GRPCClientSettings struct {
 	Endpoint string `mapstructure:"endpoint"`
 
 	// The compression key for supported compression types within collector.
-	Compression string `mapstructure:"compression"`
+	Compression middleware.CompressionType `mapstructure:"compression"`
```
Comment (bogdandrutu) [inline config/configgrpc/configgrpc.go:69]: There is a problem with this, similar in http case. We are exposing publicly an "internal" type (under internal/** path). We need to move `middleware.CompressionType` in a public package. We can do this in a separate PR, but I want you to have a tracking issue.

Comment (hyunuk) [reply config/configgrpc/configgrpc.go:69]: Posted a new issue #4627

**Review by bogdandrutu — APPROVED** [review]
